bofire/data_models/surrogates/single_task_gp.py

```diff
@@ -1,19 +1,98 @@
-from typing import Literal
+from typing import Literal, Optional
 
+import pandas as pd
 from pydantic import Field
 
-from bofire.data_models.kernels.api import AnyKernel, MaternKernel, ScaleKernel
+from bofire.data_models.domain.api import Inputs
+from bofire.data_models.enum import RegressionMetricsEnum
+from bofire.data_models.features.api import CategoricalInput
+from bofire.data_models.kernels.api import (
+    AnyKernel,
+    MaternKernel,
+    RBFKernel,
+    ScaleKernel,
+)
 from bofire.data_models.priors.api import (
     BOTORCH_LENGTHCALE_PRIOR,
     BOTORCH_NOISE_PRIOR,
     BOTORCH_SCALE_PRIOR,
+    MBO_LENGTHCALE_PRIOR,
+    MBO_NOISE_PRIOR,
+    MBO_OUTPUTSCALE_PRIOR,
     AnyPrior,
 )
+
+# from bofire.data_models.strategies.api import FactorialStrategy
 from bofire.data_models.surrogates.botorch import BotorchSurrogate
 from bofire.data_models.surrogates.scaler import ScalerEnum
+from bofire.data_models.surrogates.trainable import Hyperconfig, TrainableSurrogate
+
+
+class SingleTaskGPHyperconfig(Hyperconfig):
+    type: Literal["SingleTaskGPHyperconfig"] = "SingleTaskGPHyperconfig"
+    inputs: Inputs = Inputs(
+        features=[
+            CategoricalInput(
+                key="kernel", categories=["rbf", "matern_1.5", "matern_2.5"]
+            ),
+            CategoricalInput(key="prior", categories=["mbo", "botorch"]),
+            CategoricalInput(key="ard", categories=["True", "False"]),
+        ]
+    )
+    target_metric = RegressionMetricsEnum.MAE
+    hyperstrategy: Literal[
+        "FactorialStrategy", "SoboStrategy", "RandomStrategy"
+    ] = "FactorialStrategy"
+
+    @staticmethod
+    def _update_hyperparameters(
+        surrogate_data: "SingleTaskGPSurrogate", hyperparameters: pd.Series
+    ):
+        def matern_25(ard: bool, lenghtscale_prior: AnyPrior) -> MaternKernel:
+            return MaternKernel(nu=2.5, lengthscale_prior=lenghtscale_prior, ard=ard)
```

Comment:
@jduerholt typo here: lenghtscale_prior --> lengthscale_prior

Reply:
Thanks for spotting, I just fixed it.